app/models/concerns/profile_searching.rb

```diff
@@ -86,6 +86,26 @@ module ProfileSearching
             .or(search.where(profile_cond))
             .distinct
     }
+    scope :profile_benchmark, lambda { |benchmark_ref_id|
+      joins(:benchmark).where(benchmarks: { ref_id: benchmark_ref_id })
+    }
+    scope :latest_benchmark_profiles, lambda { |profiles|
+      latest_p = []
+      profiles.find_each do |new_p|
```

Comment:
Preliminary suggestion: Try making the scope to produce a query (another scope in a sense). Scope should return a scope that can be further used (and scoped again).

Reply:
Oh right, okay thank you!

Reply:
Note, that it does not have to be a scope on Benchmark (or scope in general). It can be a regular class method. Reduction of number of queries can apply.